In [10]:
import pandas as pd
import numpy as np
for elec_grp in ('Auditory','Sensorymotor','Motor'):
    for alignment in ('Aud','Go','Resp'):
        
        filename = f"results/{elec_grp}_{alignment}_All.csv"
        raw = pd.read_csv(filename)
        raw['vWM_lambda'] = raw['vWM_lambda'].replace(1e4, np.nan)
        raw['novWM_lambda'] = raw['novWM_lambda'].replace(1e4, np.nan) # Included for completeness, though it's likely not in this column
        df_filtered = raw[raw['time_point'] > 0]
    
        max_vWM_lambda = np.nanmedian(df_filtered['vWM_lambda'])
        max_novWM_lambda = np.nanmedian(df_filtered['novWM_lambda'])
    
        print(f"{elec_grp} in {alignment} Maximum vWM_lambda: {max_vWM_lambda}")
        print(f"{elec_grp} in {alignment} Maximum novWM_lambda: {max_novWM_lambda}")

Auditory in Aud Maximum vWM_lambda: 0.125892541179417
Auditory in Aud Maximum novWM_lambda: 0.0158489319246111
Auditory in Go Maximum vWM_lambda: 0.158489319246111
Auditory in Go Maximum novWM_lambda: 0.0316227766016838
Auditory in Resp Maximum vWM_lambda: 0.251188643150958
Auditory in Resp Maximum novWM_lambda: 0.0501187233627271
Sensorymotor in Aud Maximum vWM_lambda: 0.0158489319246111
Sensorymotor in Aud Maximum novWM_lambda: 0.0794328234724281
Sensorymotor in Go Maximum vWM_lambda: 0.0398107170553497
Sensorymotor in Go Maximum novWM_lambda: 0.158489319246111
Sensorymotor in Resp Maximum vWM_lambda: 0.0398107170553497
Sensorymotor in Resp Maximum novWM_lambda: 0.199526231496888
Motor in Aud Maximum vWM_lambda: 0.0794328234724281
Motor in Aud Maximum novWM_lambda: 0.0001
Motor in Go Maximum vWM_lambda: 0.199526231496888
Motor in Go Maximum novWM_lambda: 0.0001
Motor in Resp Maximum vWM_lambda: 0.158489319246111
Motor in Resp Maximum novWM_lambda: 0.0001


### Selected lambdas  
Auditory vWM:  0.125892541179417  
Auditory novWM: 0.0158489319246111  
Sensorymotor vWM: 0.0158489319246111  
Sensorymotor novWM: 0.0794328234724281  
Motor vWM: 0.158489319246111  
Motor novWM: 0.0001

